### PR TITLE
EAD/session6-1/Assignment1&Assignment2

### PR DESCRIPTION
**Pull request by meta-nilesh-patel**

### CODE REVIEW

File: EAD/Session 6(1)/Assignment1/demorestapi/src/main/java/com/metacube/demorestapi/repositories/InventoryRepository.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+	/**
+	 * This method search inventory in the inventory list else return new inventory.  
+	 * @param name : name of inventory to be searched.
+	 * @return inventory if found else return new inventory.
+	 */
+	public Inventory getInventory(String name) 
+	{
+		for (Inventory i : inventories) 
+		{
+			if (i.getName().equals(name))
```
Comment (rash-tiwari) [inline EAD/Session 6(1)/Assignment1/demorestapi/src/main/java/com/metacube/demorestapi/repositories/InventoryRepository.java]: we can use ```equalsIgnoreCase``` also do the similar changes at other places.

File: EAD/Session 6(1)/Assignment1/demorestapi/src/main/java/com/metacube/demorestapi/repositories/InventoryRepository.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+		int index = isInventoryExist(inv);
+		if (index == -1) 
+		{
+			inventories.add(inv);
+			return inv;
+		}
+		Inventory currentInventory = inventories.get(index);
+		int currentQuantity = currentInventory.getQuantity();
+		int newQuantity = inv.getQuantity();
+		currentInventory.setQuantity(currentQuantity + newQuantity);
```
Comment (rash-tiwari) [inline EAD/Session 6(1)/Assignment1/demorestapi/src/main/java/com/metacube/demorestapi/repositories/InventoryRepository.java]: Can modify line ```currentInventory.setQuantity(currentInventory.getQuantity()+ inv.getQuantity());```

File: EAD/Session 6(1)/Assignment1/demorestapi/src/main/java/com/metacube/demorestapi/repositories/InventoryRepository.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+		return currentInventory;
+	}
+	
+	/**
+	 * This method delete all the inventories;
+	 */
+	public List<Inventory> deleteInventories() 
+	{
+		inventories.clear();
+		return inventories;
```
Comment (rash-tiwari) [inline EAD/Session 6(1)/Assignment1/demorestapi/src/main/java/com/metacube/demorestapi/repositories/InventoryRepository.java]: can modify ```return inventories.clear();```

File: EAD/Session 6(1)/Assignment2/bookinventory/src/main/java/repositories/BookRepository.java [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+	private Connection con;
+	
+	public BookRepository() 
+	{
+		try 
+		{
+			Class.forName(driverPath);
+			con = DriverManager.getConnection(url, userName, password);
+		}
+		catch (Exception e)
```
Comment (rash-tiwari) [inline EAD/Session 6(1)/Assignment2/bookinventory/src/main/java/repositories/BookRepository.java:34]: Add specific Exception as well eg. ```ClassNotFoundException```

File: EAD/Session 6(1)/Assignment1/demorestapi/src/main/java/com/metacube/demorestapi/repositories/InventoryRepository.java (hunk 1 above)
Comment (meta-nilesh-patel) [reply EAD/Session 6(1)/Assignment1/demorestapi/src/main/java/com/metacube/demorestapi/repositories/InventoryRepository.java]: Done

File: EAD/Session 6(1)/Assignment1/demorestapi/src/main/java/com/metacube/demorestapi/repositories/InventoryRepository.java (hunk 2 above)
Comment (meta-nilesh-patel) [reply EAD/Session 6(1)/Assignment1/demorestapi/src/main/java/com/metacube/demorestapi/repositories/InventoryRepository.java]: Done

File: EAD/Session 6(1)/Assignment1/demorestapi/src/main/java/com/metacube/demorestapi/repositories/InventoryRepository.java (hunk 3 above)
Comment (meta-nilesh-patel) [reply EAD/Session 6(1)/Assignment1/demorestapi/src/main/java/com/metacube/demorestapi/repositories/InventoryRepository.java]: done

File: EAD/Session 6(1)/Assignment2/bookinventory/src/main/java/repositories/BookRepository.java (hunk 4 above)
Comment (meta-nilesh-patel) [reply EAD/Session 6(1)/Assignment2/bookinventory/src/main/java/repositories/BookRepository.java:34]: Ok, I will.

**Review by rash-tiwari — APPROVED** [review]

**Pull request merged**